x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -191,24 +191,7 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
   const anomalyThreshold = formAnomalyThreshold ?? initialState.anomalyThreshold;
   const ruleType = formRuleType || initialState.ruleType;
   const [indexPatternsLoading, { browserFields, indexPatterns }] = useFetchIndex(index);
-  const aggregatableFields = Object.entries(browserFields).reduce<BrowserFields>(
-    (groupAcc, [groupName, groupValue]) => {
-      return {
-        ...groupAcc,
-        [groupName]: {
-          fields: Object.entries(groupValue.fields ?? {}).reduce<
-            Record<string, Partial<BrowserField>>
-          >((fieldAcc, [fieldName, fieldValue]) => {
-            if (fieldValue.aggregatable === true) {
-              fieldAcc[fieldName] = fieldValue;
-            }
-            return fieldAcc;
-          }, {}),
-        } as Partial<BrowserField>,
-      };
-    },
-    {}
-  );
+  const fields: Readonly<BrowserFields> = aggregatableFields(browserFields);
```

Comment:
wouldn't it make sense to memoize this so this function is only called when the browserFields change? i.e. `const fields: Readonly<BrowserFields> = useMemo(() => aggregatableFields(browserFields), [browserFields]);`

Reply:
Yea I think it would, but I'd rather someone who is familiar with this code do that.

Reply:
This is a high confidence / low risk change that makes the UI snappy even with 14k fields. I think we should schedule additional optimization work as well.

Reply:
I think it would, but I see this commit was already merged into another branch so feel free to ignore ha